include/vara/Feature/FeatureSort.h

```diff
@@ -0,0 +1,40 @@
+#ifndef VARA_FEATURE_FEATURESORT_H
+#define VARA_FEATURE_FEATURESORT_H
+
+#include "FeatureModel.h"
+
+namespace vara::feature {
+
+class FeatureSort {
+public:
+  using feature_heap_iterator = typename std::vector<Feature *>::iterator;
+  using const_feature_heap_iterator =
+      typename std::vector<Feature *>::const_iterator;
+
+private:
+  std::vector<Feature *> Features;
+
+public:
+  template <typename T> FeatureSort(T FM) {
+    for (auto F = FM->begin(); F != FM->end(); ++F) {
+      Features.push_back(*F);
+    }
+    std::sort(Features.begin(), Features.end(),
+              Feature::FeatureDepthFirstComparator());
+  }
```

Comment:
Currently, this is not a heap. Didn't you want to make it a heap?

Reply:
Building a heap I would need to call `sort_heap` to receive a sorted collection while losing the heap property. As a custom iterator does not seem reasonable at this point, I used a  comparator with insertion at `std::upper_bound` instead.